cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/TestwellCtcTxtParser.java

```diff
@@ -79,113 +71,88 @@ public TestwellCtcTxtParser() {
     LOG.debug("Parsing 'Testwell CTC++' textual format");
-    
+
     try {
       this.scanner = new Scanner(report).useDelimiter(SECTION_SEP);
-      this.matcher = FILE_HEADER.matcher("");
-      
-      this.state = State.BEGIN;
-      parseReportHead();
-
-      while(State.PARSING == this.state) {
-          parseUnit(coverageData);
+      Matcher headerMatcher = FILE_HEADER.matcher(scanner.next());
+      while (parseUnit(coverageData, headerMatcher)) {
+        headerMatcher.reset(scanner.next());
       }
-      
     } catch (FileNotFoundException e) {
-      LOG.warn("TestwellCtcTxtParser file not found '{}'", e.getMessage());
-    }
-  }
-
-  private void parseReportHead() {
-    try {
-      if (!matcher.reset(scanner.next()).find()) {
-        LOG.info("'Testwell CTC++' file section not found.");
-      } else {
-        this.state = State.PARSING;
-      }
+      LOG.warn("File not found '{}'", e.getMessage());
     } catch (NoSuchElementException e) {
-      LOG.debug("'Testwell CTC++' file section not found!");
+      LOG.debug("File section not found.");
     }
   }
 
-  private void parseUnit(final Map<String, CoverageMeasures> coverageData) {
-    LOG.debug(matcher.toString());
-    
-    if (matcher.usePattern(FILE_HEADER).find(FROM_START)) {
-      parseFileUnit(coverageData);
+  private boolean parseUnit(final Map<String, CoverageMeasures> coverageData, Matcher headerMatcher) {
+    LOG.debug(headerMatcher.toString());
+
+    if (headerMatcher.find(FROM_START)) {
+      parseFileUnit(coverageData, headerMatcher);
     } else {
-      state = State.END;
       scanner.close();
+      return false;
     }
+    return true;
   }
 
-  private void parseFileUnit(final Map<String, CoverageMeasures> coverageData) {
+  private void parseFileUnit(final Map<String, CoverageMeasures> coverageData, Matcher headerMatcher) {
     LOG.debug("Parsing file section...");
-    
+
     String normalFilename;
-    String filename;
-    File file;
-    
-    filename = matcher.group(1);
-    if((new File(filename)).isAbsolute() == false) {
+    String filename = headerMatcher.group(1);
+    if (!(new File(filename)).isAbsolute()) {
       normalFilename = FilenameUtils.normalize("./" + filename);
     } else {
       normalFilename = FilenameUtils.normalize(filename);
     }
-    file = new File(normalFilename);
+    File file = new File(normalFilename);
     addLines(file, coverageData);
-    matcher.reset(scanner.next());
   }
 
   private void addLines(File file, final Map<String, CoverageMeasures> coverageData) {
     LOG.debug("Parsing function sections...");
 
     CoverageMeasures coverageMeasures = CoverageMeasures.create();
-    while (!matcher.reset(scanner.next()).usePattern(FILE_RESULT).find()) {
-      parseLineSection(coverageMeasures);
+    String nextLine = scanner.next();
+    while (!FILE_RESULT.matcher(nextLine).find()) {
```

Comment:
looks like a classic use-case for a `for loop` to me:
```
for (String nextLine = scanner.next(); FILE_RESULT.matcher(nextLine).find(); nextLine = scanner.next())
{
   parseLineSection(coverageMeasures, nextLine);
}
```
... but it's a matter of taste